_posts/2018-08-09-deploying-esxi-through-foreman.md

```diff
@@ -0,0 +1,402 @@
+  Now create a dummy entry for ESXi since each OS entry in Foreman needs an installation media.
+    <p float='left' align='left'>
+      <img src="/static/images/blog_images/2018-08-16-deploying-esxi-through-foreman/dummy_esxi_media.png" width="400" />
+    </p>
+
+**1.2: Create the OS**
+  * Open https://{foreman-url}/operatingsystems and click on **Create Operating System** 
+  * Under Operating System, enter the following:  
+    Name            -  ESXi-6.7.0-8169922 (ESXi-{**OS Version**}-{**Build Number**})  
+    Major version   -  6  
+    Minor version   -  7  
+    Family          -  Redhat  
+    Root pass hash  -  MD5  
+    Architectures   -  x86_64  
+    <p float='left' align='left'>
+      <img src="/static/images/blog_images/2018-08-16-deploying-esxi-through-foreman/new_esxi_os.png" width="400" />
+    </p>
+  * Partition Table - Kickstart default  
+    Note: We are using Kickstart default for the sake of simplicity. The actual partitioning will be controlled from the kickstart file. It is also possible to create a new ptable for ESXi and use it here.
+    <p float='left' align='left'>
+      <img src="/static/images/blog_images/2018-08-16-deploying-esxi-through-foreman/new_esxi_os_ptable.png" width="400" />
+    </p>
+  * Installation Media - Select the dummy media created above
+    <p float='left' align='left'>
+      <img src="/static/images/blog_images/2018-08-16-deploying-esxi-through-foreman/new_esxi_os_media.png" width="400" />
+    </p>
+  * Press Submit
+
+**1.3: Create provisioning templates**
+  * **1.3.1:** Create the PXELinux template
+    * Open https://{foreman-url}/templates/provisioning_templates and click on **Create Template**
+    * Name - ESXi-6.7.0-8169922
+    * Template contents:
+
+      ```
+      DEFAULT ESXi
+      NOHALT 1
+      LABEL ESXi
+      KERNEL boot/ESXi-6.7.0-8169922/mboot.c32
+      APPEND -c boot-ESXi-6.7.0-8169922.cfg
+      IPAPPEND 2
+      ```
+    <p float='left' align='left'>
+      <img src="/static/images/blog_images/2018-08-16-deploying-esxi-through-foreman/esxi_pxelinux_template_new.png" width="400" />
+    </p>
+      Note the **KERNEL** and **APPEND** lines. These will be useful when we actually mount the ISO image.
+    * Template Type - **PXELinux Template**
+      <p float='left' align='left'>
+        <img src="/static/images/blog_images/2018-08-16-deploying-esxi-through-foreman/esxi_pxelinux_template_new_type.png" width="400" />
+      </p>
+    * Template Association - Select **"ESXi 6.7 Build 8169922"**
+      <p float='left' align='left'>
+        <img src="/static/images/blog_images/2018-08-16-deploying-esxi-through-foreman/esxi_pxelinux_template_new_assn.png" width="400" />
+      </p>
+    * Click Submit
+
+  * **1.3.2:** Create the Kickstart Template
+    * Click on **Create Template**
+    * Name - ESXi Minimal Kickstart
+    * A sample template:  
+      ```
+      vmaccepteula
+      keyboard 'US Default'
+      reboot
+      rootpw P@ssw0rd
```

Comment:
Would it be possible to use the password macro for this or does it have to be plain text?

Reply:
Removed plaintext password and used the macro now.